crates/trie/parallel/src/proof_task.rs

```diff
@@ -265,19 +268,37 @@ where
         );
 
         let (trie_cursor_factory, hashed_cursor_factory) = self.create_factories();
+        let multi_added_removed_keys = input
+            .multi_added_removed_keys
+            .unwrap_or_else(|| Arc::new(MultiAddedRemovedKeys::new()));
+        let added_removed_keys = multi_added_removed_keys.get_storage(&input.hashed_address);
+
+        let span = tracing::trace_span!(
+            target: "trie::proof_task",
+            "Storage proof calculation",
+            hashed_address=?input.hashed_address,
+            // Add a random id because we often have parallel storage proof calculations for the
+            // same hashed address, and we want to differentiate them during trace analysis.
+            span_id=rand::random::<u64>(),
```

Comment:
I would prefer not to be doing `rand` operations here, maybe we could just assign each task a number for this?